lib/token-store/index.js

```diff
@@ -549,4 +550,71 @@ module.exports = class TokenStore {
             padding
         ).getAllTokens();
     }
+
+    /**
+     * Gets all comments before the given node or token.
+     * @param {ASTNode|token} nodeOrToken The AST node or token to get the comments for.
+     * @returns {Object} The list of comments indexed by their position.
+     * @public
+     */
+    getCommentsBefore(nodeOrToken) {
+        const tokens = [];
+        const cursor = createCursorWithCount(
+            cursors.backward,
+            this[TOKENS],
+            this[COMMENTS],
+            this[INDEX_MAP],
+            -1,
+            nodeOrToken.range[0],
+            { includeComments: true }
+        );
+        let currentToken = cursor.getOneToken();
+
+        while (currentToken && astUtils.isCommentToken(currentToken)) {
+            tokens.unshift(currentToken);
```

Comment:
I've heard using `push` followed by `reverse` is faster than using `unshift` for large arrays, but I'm not sure whether that's true in this case or if it matters.